go/kbcmf/encrypt_test.go

```diff
@@ -0,0 +1,1093 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package kbcmf
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"golang.org/x/crypto/nacl/box"
+	"testing"
+)
+
+type boxPublicKey struct {
+	key RawBoxKey
+}
+
+type boxSecretKey struct {
+	pub boxPublicKey
+	key RawBoxKey
+}
+
+type keyring struct {
+	keys map[string]BoxSecretKey
+}
+
+func newKeyring() *keyring {
+	return &keyring{
+		keys: make(map[string]BoxSecretKey),
+	}
+}
+
+func (r *keyring) insert(k BoxSecretKey) {
+	r.keys[hex.EncodeToString(k.GetPublicKey().ToKID())] = k
+}
+
+func (r *keyring) LookupBoxPublicKey(kid []byte) BoxPublicKey {
+	priv := r.keys[hex.EncodeToString(kid)]
+	if priv == nil {
+		return nil
+	}
+	return priv.GetPublicKey()
+}
+
+func (r *keyring) LookupBoxSecretKey(kids [][]byte) (int, BoxSecretKey) {
+	for i, kid := range kids {
+		if key, _ := r.keys[hex.EncodeToString(kid)]; key != nil {
+			return i, key
+		}
+	}
+	return -1, nil
+}
+
+func (b boxPublicKey) ToRawBoxKeyPointer() *RawBoxKey {
+	return &b.key
+}
+
+func (b boxPublicKey) ToKID() []byte {
+	ret := append([]byte{0x01, 0x21}, b.key[:]...)
+	ret = append(ret, byte(0x0a))
+	return ret
```

Comment:
I'd say drop these tags for now. If we want to add support for more types of keys, we'll probably increment the version number anyway. But again, not blocking the initial checkin.